replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicatorHistoryImpl.java

```diff
@@ -134,8 +215,86 @@ private Metacard getMetacardFromStatus(ReplicationStatus replicationStatus) {
     helper.setIfPresent(mcard, ReplicationHistory.PULL_BYTES, replicationStatus.getPullBytes());
     helper.setIfPresent(mcard, ReplicationHistory.PUSH_BYTES, replicationStatus.getPushBytes());
     helper.setIfPresent(mcard, ReplicationHistory.STATUS, replicationStatus.getStatus().name());
+    helper.setIfPresent(mcard, ReplicationHistory.LAST_SUCCESS, replicationStatus.getLastSuccess());
+    helper.setIfPresent(mcard, ReplicationHistory.LAST_RUN, replicationStatus.getLastRun());
     mcard.setId(replicationStatus.getId());
     mcard.setTags(Collections.singleton(ReplicationHistory.METACARD_TAG));
     return mcard;
   }
+
+  private void addStats(ReplicationStatus base, ReplicationStatus status) {
+    base.setPushCount(base.getPushCount() + status.getPushCount());
+    base.setPushBytes(base.getPushBytes() + status.getPushBytes());
+    base.setPushFailCount(base.getPushFailCount() + status.getPushFailCount());
+    base.setPullCount(base.getPullCount() + status.getPullCount());
+    base.setPullBytes(base.getPullBytes() + status.getPullBytes());
+    base.setPullFailCount(base.getPullFailCount() + status.getPullFailCount());
+    if (base.getLastRun() == null || status.getStartTime().after(base.getLastRun())) {
+      base.setLastRun(status.getStartTime());
+    }
+    base.setStatus(status.getStatus());
+    if (base.getStartTime().after(status.getStartTime())) {
+      base.setStartTime(status.getStartTime());
+    }
+    if (status.getStatus().equals(Status.SUCCESS)) {
+      if (status.getPushCount() > 0 || status.getPullCount() > 0) {
+        base.setDuration(base.getDuration() + status.getDuration());
```

Comment:
❓From the API, we will expose this as "total run time" for the configuration, but would something like "last run duration" be more useful, ie don't accumulate these? I'm thinking the more runs, the less meaningful "total run time" becomes.

Reply:
I was thinking this would be useful still for calculating the average transfer rate but you have a good point that it might be also good to know what the last run duration was. Maybe also last bytes/item transferred. Would you be ok with exploring that in a later change?

Reply:
Yeah I'm okay with changing it later. I'm not sure if average transfer rate would be useful either though. I'd think they would want to know the transfer rate of the currently running job.

Reply:
❓ Shouldn't we add the duration time even if it failed? In a failure, wouldn't there still have been records replicated potentially?